Makefile

```diff
@@ -1,4 +1,7 @@
 ifneq ($O,)
+ifneq ($(shell echo $(O) | grep -- "/\.$$"),)
+$(error Cannot support output directory 'O' path ending with terminal subpath info "./")
+endif
```

Comment:
Your complaining about `./` but you're testing against `/.`, what is the correct one?
Also the test could be simplified to:
```
ifeq ($(patsubst %/.,,$(O)),)
```